### ISSUE TITLE
Test Helper Support

### ISSUE THREAD

**Issue by fsmanuel** [funkensturm/ember-cli-file-picker#34]
http://nerds.intuo.io/2016/05/12/mocking-file-uploads-in-javascript.html?utm_source=Ember+Weekly&utm_campaign=e3bc6b917d-Ember_Weekly_Issue_159&utm_medium=email&utm_term=0_e96229d21d-e3bc6b917d-106360253

https://medium.com/@chrisdmasters/acceptance-testing-file-uploads-in-ember-2-5-1c9c8dbe5368#.kol75ql51

https://github.com/201-created/ember-cli-acceptance-test-helpers/blob/master/blueprints/ember-cli-acceptance-test-helpers/index.js

related #28 #24 

fix:

```
// beta
not ok 11 PhantomJS 1.9 - Integration | Component | file picker: it's testable
        actual: >
            null
        message: >
            Assertion after the final `assert.async` was resolved
```

```
// canary
not ok 2 PhantomJS 1.9 - Acceptance | index: Works for readAsFile
    ---
        actual: >

        expected: >
            12
not ok 3 PhantomJS 1.9 - Acceptance | index: Works for readAsText
    ---
        actual: >

        expected: >
            12
```

**Comment by jelhan** [comment]
Error in beta is a little bit tricky since it doesn't fail all time. Compare these two: https://travis-ci.org/funkensturm/ember-cli-file-picker/builds/108580080 vs https://travis-ci.org/funkensturm/ember-cli-file-picker/jobs/108739541

But maybe just adding a assertion in [integration test](https://github.com/funkensturm/ember-cli-file-picker/blob/master/tests/integration/components/file-picker-test.js) might be enough.

**Comment by jelhan** [comment]
`uploadFile` helper is broken since ember 2.5. It throws:

```
ember.debug.js:38471 TypeError: Cannot assign to read only property 'target' of object '#<Event>'
    at Function.jQuery.extend.jQuery.fn.extend (jquery.js:254)
    at buildBasicEvent (ember.debug.js:37404)
    at Object.fireEvent (ember.debug.js:37394)
    at Class.triggerEvent (ember.debug.js:38108)
    at ember.debug.js:37621
    at isolate (ember.debug.js:38749)
    at ember.debug.js:38705
    at tryCatch (ember.debug.js:58173)
    at invokeCallback (ember.debug.js:58185)
    at ember.debug.js:58254
```

If tests are run in cli (or travis) you will only see an `Assertion after the final `assert.async` was resolved` error. Have to run it in browser and get actual error in console.

I'm quite sure this commit breaks our test helper: https://github.com/emberjs/ember.js/commit/3fbdcb73ae4a5fde0c7110edc37721639105ade1 It changed `triggerEvent` and introduced `buildBasicEvent` function. Test helper throws after calling `jQuery.extend(event, options);` in `buildBasicEvent`. `options` passed to `jQuery.extend` is the object hash [created in our test helper](https://github.com/funkensturm/ember-cli-file-picker/blob/master/test-support/helpers/ember-cli-file-picker.js#L35).

**Comment by fsmanuel** [comment]
@jelhan I merged #139 can we close this now?

**Comment by jelhan** [comment]
@fsmanuel I'm not using ember-cli-file-picker much these days but I think it should be fixed. If not, it could be reopened.
